<launch>

<arg name = "StartRec" default = "false" />

<arg name="start" />
<arg name="destination" />
  
<!-- Detection node -->
<node pkg="line_follower_turtlebot" name="detection" type="detect" cwd="node" output="screen">
</node>

<!-- Robot commands node -->
<node pkg="line_follower_turtlebot" name="Velocity" type="navig" cwd="node" output="screen">
</node>


<node pkg="line_follower_turtlebot" name="mode_decisioner" type="mode_decision.py" cwd="node" output="screen">
</node>
  
<node pkg="line_follower_turtlebot" name="obstacle_avoidance" type="obstacle_avoidance.py" cwd="node" output="screen">
</node>  

<node pkg="line_follower_turtlebot" name="turn_action_server" type="turn_action_server.py" cwd="node" output="screen">
</node>  


<node pkg="line_follower_turtlebot" name="turn_action_client" type="turn_action_client.py" cwd="node" output="screen">
</node> 

  
<node pkg="line_follower_turtlebot" name="path_planning" type="path_planning.py" cwd="node" output="screen" required="true">
  <arg name="$(arg start)" />
  <arg name="$(arg destination)" />
</node>   
  

<node
pkg = "rosbag"
type = "record"
name = "rosbag_record_all"
respawn = "true"
output = "screen"
args = "-a -O BagFile -x /camera/(.*) "
if = "$(arg StartRec)"
/>

</launch>
